src/jenkins_job_insight/diagnostic_archive.py

```
@@ -0,0 +1,476 @@
+    # Strip common accidental prefixes — the API endpoint already includes /artifact/
+    for prefix in ("artifact/", "artifacts/"):
+        if artifact_path.startswith(prefix):
+            artifact_path = artifact_path[len(prefix) :]
+            logger.debug(
+                f"Stripped '{prefix}' prefix from artifact path: {artifact_path}"
+            )
+            break
+
+    job_path = "/job/".join(job_name.split("/"))
+    url = f"{jenkins_url.rstrip('/')}/job/{job_path}/{build_number}/artifact/{artifact_path}"
+    logger.info(f"Downloading artifact: {url}")
```

Ruff (0.15.5)

[warning] Logging statement uses f-string

(G004)